client/ayon_core/hosts/substancepainter/api/lib.py

```diff
@@ -640,3 +642,88 @@ def _setup_prompt():
         return
 
     return project_mesh
+
+
+def get_export_presets_by_filtering(export_preset_name, channel_type_list):
+    """Function to get export presets included with specific channels
+    requested by users.
+
+    Args:
+        export_preset_name (str): Name of export preset
+        channel_type_list (list): A list of channel type requested by users
+
+    Returns:
+        dict: export preset data
+    """
+
+    new_maps = []
+
+    export_presets = get_export_presets()
+    export_preset_nice_name = export_presets[export_preset_name]
+    resource_presets = substance_painter.export.list_resource_export_presets()
+    preset = next(
+        (
+            preset for preset in resource_presets
+            if preset.resource_id.name == export_preset_nice_name
+        ), None
+    )
+    if preset is None:
+        return {}
+    maps = preset.list_output_maps()
+    for channel_map in maps:
+        for n in channel_type_list:
+            if not channel_map.get("fileName"):
+                continue
+
+            if n in channel_map["fileName"]:
+                new_maps.append(channel_map)
+    # Create a new preset
+    return {
+        "exportPresets": [
+            {
+                "name": export_preset_name,
+                "maps": new_maps
+            }
+        ],
+    }
+
+
+@contextlib.contextmanager
+def set_layer_stack_opacity(node_ids, channel_type):
```

Comment:
```suggestion
def set_layer_stack_opacity(node_ids, channel_types):
```
Make sure it's plural if it's a list.